src/core/qgsrendercontext.cpp

```diff
@@ -295,6 +295,10 @@ QgsRenderContext QgsRenderContext::fromMapSettings( const QgsMapSettings &mapSet
   ctx.mFrameRate = mapSettings.frameRate();
   ctx.mCurrentFrame = mapSettings.currentFrame();
 
+  const QStringList layerIds = mapSettings.layerIds( true );
+  if ( !layerIds.empty() )
+    ctx.setCustomRenderingFlag( QStringLiteral( "visible_layer_ids" ), layerIds );
```

Comment:
For the record, that feels like a badly named API function here. I would _not_ assume a customRendringFlags function would return a QVariantMap holding random properties :)

It would be good to possibly deprecate these functions in favor of a renamed customProperties() / setCustomProperty() / clearCustomProperties() set of functions. It should at least be flagged as a TODO 4.0: rename :)

Reply:
Ok, done!